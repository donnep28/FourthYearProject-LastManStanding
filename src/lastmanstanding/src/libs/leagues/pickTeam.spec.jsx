import React from 'react';
import {mount, shallow} from 'enzyme';
import PickTeam from './pickTeam';

const mockSetSelectedTeam = jest.fn();

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: (initial) => [initial, mockSetSelectedTeam],
}));

describe('pickTeam', () => {
  beforeAll(() => {
    jest.resetAllMocks();
  });

  const props = {
    setPick: jest.fn(),
    sub: 'mockSub',
    leagueID: 'mockLeagueId',
    teams: [
      'Manchester United FC',
      'Manchester City FC',
      'Liverpool FC',
      'Everton FC',
      'Chelsea FC',
      'Southampton FC',
      'West Ham United FC',
      'Arsenal FC',
      'Aston Villa FC',
      'Crystal Palace FC',
      'Newcastle United FC',
      'Brighton & Hove Albion FC',
      'Burnley FC',
      'Fulham FC',
      'West Bromwich Albion FC',
    ],
    setPermPick: jest.fn(),
  };
  describe('Snapshot', () => {
    it('renders teams', () => {
      const component = shallow(
          <PickTeam {...props}/>,
      );
      expect(component).toMatchSnapshot();
    });
  });
  describe('submit Team', () => {
    it('selects correct team', () => {
      const component = shallow(
          <PickTeam {...props}/>,
      );
      component.find(
          '[data-automation="Manchester United FC"]',
      ).simulate('click');
      expect(mockSetSelectedTeam).toHaveBeenCalledWith('Manchester United FC');
    });
    it('sets submitToggle to true', () => {
      const setSubmitToggle = jest.fn();
      const component = mount(<PickTeam {...props}/>);
      const handleClick = jest.spyOn(React, 'useState');
      handleClick.mockImplementation((submitToggle) =>
        [submitToggle, setSubmitToggle]);
      component.find(
          '[data-automation="Manchester United FC"]',
      ).at(0).simulate('click');
      expect(setSubmitToggle).toBeTruthy();
    });
  });
});
