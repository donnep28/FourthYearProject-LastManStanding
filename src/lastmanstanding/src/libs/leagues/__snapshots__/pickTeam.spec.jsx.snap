// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pickTeam Snapshot renders teams 1`] = `
<div>
  <div>
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="column"
      spacing={1}
    >
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Manchester United FC"
        disableRipple={true}
        key="Manchester United FC"
        onClick={[Function]}
        variant="contained"
      >
        Manchester United FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Manchester City FC"
        disableRipple={true}
        key="Manchester City FC"
        onClick={[Function]}
        variant="contained"
      >
        Manchester City FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Liverpool FC"
        disableRipple={true}
        key="Liverpool FC"
        onClick={[Function]}
        variant="contained"
      >
        Liverpool FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Everton FC"
        disableRipple={true}
        key="Everton FC"
        onClick={[Function]}
        variant="contained"
      >
        Everton FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Chelsea FC"
        disableRipple={true}
        key="Chelsea FC"
        onClick={[Function]}
        variant="contained"
      >
        Chelsea FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Southampton FC"
        disableRipple={true}
        key="Southampton FC"
        onClick={[Function]}
        variant="contained"
      >
        Southampton FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="West Ham United FC"
        disableRipple={true}
        key="West Ham United FC"
        onClick={[Function]}
        variant="contained"
      >
        West Ham United FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Arsenal FC"
        disableRipple={true}
        key="Arsenal FC"
        onClick={[Function]}
        variant="contained"
      >
        Arsenal FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Aston Villa FC"
        disableRipple={true}
        key="Aston Villa FC"
        onClick={[Function]}
        variant="contained"
      >
        Aston Villa FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Crystal Palace FC"
        disableRipple={true}
        key="Crystal Palace FC"
        onClick={[Function]}
        variant="contained"
      >
        Crystal Palace FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Newcastle United FC"
        disableRipple={true}
        key="Newcastle United FC"
        onClick={[Function]}
        variant="contained"
      >
        Newcastle United FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Brighton & Hove Albion FC"
        disableRipple={true}
        key="Brighton & Hove Albion FC"
        onClick={[Function]}
        variant="contained"
      >
        Brighton & Hove Albion FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Burnley FC"
        disableRipple={true}
        key="Burnley FC"
        onClick={[Function]}
        variant="contained"
      >
        Burnley FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="Fulham FC"
        disableRipple={true}
        key="Fulham FC"
        onClick={[Function]}
        variant="contained"
      >
        Fulham FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
      <WithStyles(WithStyles(ForwardRef(Button)))
        className="makeStyles-margin-1"
        data-automation="West Bromwich Albion FC"
        disableRipple={true}
        key="West Bromwich Albion FC"
        onClick={[Function]}
        variant="contained"
      >
        West Bromwich Albion FC
      </WithStyles(WithStyles(ForwardRef(Button)))>
    </WithStyles(ForwardRef(Grid))>
  </div>
  <div>
    <SelectTeam
      leagueID="mockLeagueId"
      selectedTeam=""
      setPermPick={[MockFunction]}
      sub="mockSub"
      submitToggle={false}
    />
  </div>
</div>
`;
